flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/metrics/MetricAggregator.java

```diff
@@ -34,7 +33,11 @@ public enum MetricAggregator {
         this.getter = getter;
     }
 
-    public Optional<Double> get(AggregatedMetric metric) {
-        return Optional.ofNullable(metric).map(getter).filter(d -> !d.isNaN());
+    public double get(AggregatedMetric metric) {
+        if (metric != null) {
+            return getter.apply(metric);
+        } else {
+            return Double.NaN;
+        }
```

Comment:
This change here could be the reason why the lag growth rate may be nan after this.
Before this commit after your pending record related change the lag would be 0 , and now this may return Double.nan.

Reply:
We discussed that this may delay decisions in case of NaNs but should not cause issues.